appinventor/blocklyeditor/src/field_lexical_variable.js

```diff
@@ -675,17 +675,20 @@ Blockly.LexicalVariable.renameParamWithoutRenamingCapturables = function (source
 //  }
   var sourcePrefix = "";
   if (Blockly.showPrefixToUser) {
-    if (sourceBlock.type == "procedures_mutatorarg"
-        || sourceBlock.type == "procedures_defnoreturn"
-        || sourceBlock.type == "procedures_defreturn") {
+    var type = sourceBlock.type;
+    if (type == "procedures_mutatorarg"
+        || type == "procedures_defnoreturn"
+        || type == "procedures_defreturn") {
       sourcePrefix = Blockly.procedureParameterPrefix;
-    } else if (sourceBlock.type == "controls_forEach") {
+    } else if (type == "controls_forEach") {
       sourcePrefix = Blockly.loopParameterPrefix;
-    } else if ( sourceBlock.type == "controls_forRange") {
+    } else if (type == "controls_for_each_dict") {
+      sourcePrefix = Blockly.loopKeyParameterPrefix;
```

Comment:
Should I handle this by combining the loopKeyParameterPrefix and loopValueParameterPrefix into one loopDictParameter prefix? Or should I try and use the sourceBlock to figure out which one is being renamed?

Reply:
I would say keep them separate for now. An earlier version of App inventor differentiated different types of variables based on their scopes, which is why the distinction is made here. Today, the only variable type that is differentiated is global variables. If we were ever to go back to differentiating different variable types then having this here would save us the work.